shared/api/cluster.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+// Cluster represents high-level information about a LXD cluster.
+type Cluster struct {
+	StoragePools []StoragePool `json:"storage_pools" yaml:"storage_pools"`
+	Networks     []Network     `json:"networks" yaml:"networks"`
+}
+
+// ClusterPost represents the fields required to bootstrap or join a LXD
+// cluster.
+//
+// API extension: cluster
+type ClusterPost struct {
+	Name           string        `json:"name" yaml:"name"`
+	Address        string        `json:"address" yaml:"address"`
+	Schema         int           `json:"schema" yaml:"schema"`
+	API            int           `json:"api" yaml:"api"`
+	TargetAddress  string        `json:"target_address" yaml:"target_address"`
+	TargetCert     string        `json:"target_cert" yaml:"target_cert"`
+	TargetCA       []byte        `json:"target_ca" yaml:"target_ca"`
+	TargetPassword string        `json:"target_password" yaml:"target_password"`
+	StoragePools   []StoragePool `json:"storage_pools" yaml:"storage_pools"`
+	Networks       []Network     `json:"networks" yaml:"networks"`
+}
+
+// ClusterNodeAccepted represents the response of a request to join a cluster.
+//
+// API extension: cluster
+type ClusterNodeAccepted struct {
+	RaftNodes  []RaftNode `json:"raft_nodes" yaml:"raft_nodes"`
+	PrivateKey []byte     `json:"private_key" yaml:"private_key"`
+}
+
+// RaftNode represents the a LXD node that is part of the dqlite raft cluster.
+//
+// API extension: cluster
+type RaftNode struct {
```

Comment:
`ClusterRaftNodes`

Reply:
I'm not crazy about having raft be mentioned in our public API, especially when there doesn't really appear to be much of anything raft specific there. Is the Address there the same as the URL for the member?

If so, how about we just add the ID to ClusterMember and have the join response include a full list of all members, would that get us what you need?

Reply:
Not really. This list must hold essentially two things 1) IP addresses of database nodes (which yes, is roughly the same as ClusterMember.URL) 2) their raft server ID (this ID is a raft internal thing, unrelated to our database ID for ClusterMember).

We could probably rename RaftNode to ClusterDatabaseMember or something like that, if you wish.

Reply:
How about adding a DatabaseID to the ClusterMember struct which would contain that raft node id when Database is true?

Would that then let you use a []ClusterMember when joining? Having this information visible in the API would also make it easier to debug some logs if the raft node id shows up in there.

Reply:
This has been moved to an internal API, so it's not anymore public.